sdcm/nemesis.py

```diff
@@ -434,6 +437,135 @@ def disrupt_stop_start_scylla_server(self):  # pylint: disable=invalid-name
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        trigger_func = {"func": compaction_ops.trigger_major_compaction, "kwargs": {}}
+        watch_func = {"func": compaction_ops.stop_on_user_compaction_logged,
+                      "kwargs": {"node": node,
+                                 "watch_for": "User initiated compaction",
+                                 "timeout": 120,
+                                 "stop_func": compaction_ops.stop_major_compaction}}
+        self._trigger_and_stop_compaction(trigger_func_w_kwargs=trigger_func, watch_func_w_kwargs=watch_func)
+
+    def disrupt_start_stop_scrub_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        trigger_func = {"func": compaction_ops.trigger_scrub_compaction, "kwargs": {}}
+        watcher_func = {"func": compaction_ops.stop_on_user_compaction_logged,
+                        "kwargs": {"node": node,
+                                   "watch_for": "Scrubbing",
+                                   "timeout": 120,
+                                   "stop_func": compaction_ops.stop_scrub_compaction}}
+        self._trigger_and_stop_compaction(trigger_func_w_kwargs=trigger_func, watch_func_w_kwargs=watcher_func)
+
+    def disrupt_start_stop_cleanup_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        trigger_func = {"func": compaction_ops.trigger_cleanup_compaction, "kwargs": {}}
+        watch_func = {"func": compaction_ops.stop_on_user_compaction_logged,
+                      "kwargs": {"node": node,
+                                 "timeout": 120,
+                                 "watch_for": "Cleaning",
+                                 "stop_func": compaction_ops.stop_cleanup_compaction}}
+        self._trigger_and_stop_compaction(trigger_func_w_kwargs=trigger_func, watch_func_w_kwargs=watch_func)
+
+    def disrupt_start_stop_validation_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+        trigger_func = {"func": compaction_ops.trigger_validation_compaction, "kwargs": {}}
+        watch_func = {"func": compaction_ops.stop_on_user_compaction_logged,
+                      "kwargs": {"node": node,
+                                 "watch_for": "Scrubbing in validate mode",
+                                 "timeout": 120,
+                                 "stop_func": compaction_ops.stop_validation_compaction}}
+        self._trigger_and_stop_compaction(trigger_func_w_kwargs=trigger_func, watch_func_w_kwargs=watch_func)
+
+    def disrupt_start_stop_upgrade_compaction(self):
+        node1 = self.cluster.nodes[0]
+        upgraded_configuration_options = {"enable_sstables_mc_format": False,
+                                          "enable_sstables_md_format": True}
+
+        def _upgrade_sstables_format(node: BaseNode):
```

Comment:
Is it upgrade or Downgrade ? log string is not in line with function name

Reply:
Yeah, the original approach here was to downgrade and then upgrade the sstable format, hence the mix here. Fixed now.